File: src/kll/FallBehaviour.cpp
//
// Created by Damian Stewart on 11/12/2016.
//

#include "FallBehaviour.h"
#include "Object.h"

kll::FallBehaviour::FallBehaviour(kll::Object *target) : Behaviour(target)
{

}

void kll::FallBehaviour::Update(float dt)
{
    const vec3 GRAVITY(0, 1, 0);
    mTarget->SetVelocity(mTarget->GetVelocity() + dt * GRAVITY);

}

File: bin/data/FallBehaviour.lua
--
-- Created by IntelliJ IDEA.
-- User: damian
-- Date: 31/12/16
-- Time: 13:48
-- To change this template use File | Settings | File Templates.
--


FallBehaviour = {}

function FallBehaviour:new(target, gravity)
    o = {mTarget=target, gravity=gravity}
    setmetatable(o, self)
    self.__index = self
    return o
end

function FallBehaviour:Update(dt)

    local velocity = self.mTarget:GetVelocity()
    local acceleration = self.gravity * dt
    velocity = velocity + acceleration

    self.mTarget:SetVelocity(velocity)

end

function FallBehaviour:ShouldObjectBeDestroyed()
    return false
end

function FallBehaviour:GetTarget()
    return self.mTarget
end

